jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -169,20 +169,25 @@ object XGBoost extends Serializable {
       trainingData: Dataset[_],
       params: Map[String, Any],
       round: Int,
-      nWorkers: Int,
+      nWorkers: Int = 0,
       obj: ObjectiveTrait = null,
       eval: EvalTrait = null,
       useExternalMemory: Boolean = false,
       missing: Float = Float.NaN,
       featureCol: String = "features",
       labelCol: String = "label"): XGBoostModel = {
-    require(nWorkers > 0, "you must specify more than 0 workers")
+    val nWorkers_ = if (nWorkers == 0) {
```

Comment:
Underscore doesn't seem a good naming convention for scala:
https://github.com/databricks/scala-style-guide#variable-naming

Reply:
Thanks, how about `nWorkersInferred`?